README.md

```diff
@@ -2,92 +2,707 @@
 
+## Table of Contents
+
+- [Overview](#overview)
+- [Features](#features)
+- [Limitations](#limitations)
+- [Installation](#installation)
+- [Quick Start](#quick-start)
+  - [Basic Usage](#basic-usage)
+  - [Using the OSDF Scheme](#using-the-osdf-scheme)
+- [Object Operations](#object-operations)
+  - [Listing Objects and Collections](#listing-objects-and-collections)
+  - [Pattern Matching with Glob](#pattern-matching-with-glob)
+  - [Reading Objects](#reading-objects)
+  - [Writing Objects](#writing-objects)
+  - [Downloading Objects](#downloading-objects)
+- [Advanced Configuration](#advanced-configuration)
+  - [Specifying Endpoints](#specifying-endpoints)
+  - [Enabling Direct Reads](#enabling-direct-reads)
+  - [Specifying Preferred Caches](#specifying-preferred-caches)
+- [Authorization](#authorization)
+  - [1. Providing a Token via Headers](#1-providing-a-token-via-headers)
+  - [2. Environment Variables](#2-environment-variables)
+  - [3. Default Token Location](#3-default-token-location)
+  - [4. HTCondor Token Discovery](#4-htcondor-token-discovery)
+  - [Token File Formats](#token-file-formats)
+  - [Automatic Token Discovery](#automatic-token-discovery)
+  - [Token Scopes](#token-scopes)
+  - [Token Validation](#token-validation)
+- [Integration with Data Science Libraries](#integration-with-data-science-libraries)
+  - [Using with xarray and Zarr](#using-with-xarray-and-zarr)
+  - [Using with PyTorch](#using-with-pytorch)
+  - [Using with Pandas](#using-with-pandas)
+- [Getting an FSMap](#getting-an-fsmap)
+- [Monitoring and Debugging](#monitoring-and-debugging)
+  - [Access Statistics](#access-statistics)
+  - [Enabling Debug Logging](#enabling-debug-logging)
+- [API Reference](#api-reference)
+  - [PelicanFileSystem](#pelicanfilesystem)
+  - [OSDFFileSystem](#osdffilesystem)
+  - [PelicanMap](#pelicanmap)
+- [Examples](#examples)
+- [Troubleshooting](#troubleshooting)
+- [Contributing](#contributing)
+- [License](#license)
+- [Citation](#citation)
+- [Support](#support)
+
 ## Overview
 
-PelicanFS is a file system interface (fsspec) for the Pelican Platform.  For more information about pelican, see our [main website](https://pelicanplatform.org) or [Github page](https://github.com/PelicanPlatform/pelican). For more information about fsspec, visit the [filesystem-spec](https://filesystem-spec.readthedocs.io/en/latest/index.html) page.
+PelicanFS is a file system interface (fsspec) for the Pelican Platform. It provides a Python interface to interact with Pelican federations, allowing you to read, write, and manage objects across distributed object storage systems.
+
+For more information about Pelican, see our [main website](https://pelicanplatform.org) or [GitHub page](https://github.com/PelicanPlatform/pelican). For more information about fsspec, visit the [filesystem-spec](https://filesystem-spec.readthedocs.io/en/latest/index.html) page.
```

Comment:
Maybe link to the pelican docs in addition to the main landing page? I view `https://pelicanplatform.org` as fluff to keep some funding org happy -- it's full of buzzwords, but little actual content to explain what Pelican is or how it works.